src/Utilities/TaskItem.cs

```diff
@@ -27,7 +27,7 @@ namespace Microsoft.Build.Utilities
     /// <comment>
     /// Surprisingly few of these Utilities TaskItems are created: typically several orders of magnitude fewer than the number of engine TaskItems.
     /// </comment>
-    public sealed class TaskItem : MarshalByRefObject, ITaskItem, ITaskItem2
+    public sealed class TaskItem : MarshalByRefObject, ITaskItem, ITaskItem2, IEquatable<TaskItem>
```

Comment:
It would be nice to move this change to `TaskItem` to a separate commit and add explicit `TaskItem` tests for this, not related to the `Copy` task.